framework/Optimizers/Optimizer.py

```diff
@@ -571,12 +596,40 @@ def updateMultilevelDepth(self, traj, depth, optPoint, setAll=False):
     # clear existing gradient determination data
     if not firstTime:
       self.clearCurrentOptimizationEffort(traj)
+    # apply preconditioner IFF we're going towards INNER loops
+    if depth > oldDepth:
+      self.raiseADebug('Preconditioning subsets below',oldDepth,range(oldDepth+1,depth+1))
+      #apply changes all the way down
+      for d in range(oldDepth+1,depth+1):
+        precondBatch = self.mlSequence[d]
+        precond = self.mlPreconditioners.get(precondBatch,None)
+        if precond is not None:
+          self.raiseADebug('Running preconditioner on batch "{}"'.format(precondBatch))
+          infoDict = {'SampledVars':self.denormalizeData(optPoint)}
+          _,(results,_) = precond.evaluateSample([infoDict['SampledVars']],'Optimizer',infoDict)
+          # flatten results #TODO breaks for multi-entry arrays
```

Comment:
The objects returned currently are always single-valued, as [1.234].  Multi-entry I intend to convey arrays with more than 1 entry.